rayon-core/src/registry.rs

```diff
@@ -87,20 +87,21 @@ unsafe fn init_registry(config: Configuration) {
 
 impl Registry {
     pub fn new(mut configuration: Configuration) -> Arc<Registry> {
-        debug_assert!(configuration.validate().is_ok());
-
-        let limit_value = match configuration.num_threads() {
-            Some(value) => value,
-            None => {
-                match env::var("RAYON_RS_NUM_CPUS") {
-                    Ok(s) => usize::from_str(&s).expect("invalid value for RAYON_RS_NUM_CPUS"),
-                    Err(_) => num_cpus::get(),
-                }
+        // Determine number of threads to spawn. Use configuration value if not zero.
+        // If configuration is zero try the RAYON_RS_NUM_CPUS env var.
+        // If that fails use the default (num_cpus).
+        let config_num_threads = configuration.num_threads();
```

Comment:
With the logic moved, we can directly use this value.  Maybe just `assert` that we ended up non-zero.